fe/fe-core/src/main/java/org/apache/doris/catalog/RefreshManager.java

```diff
@@ -146,4 +159,45 @@ private void refreshInternalCtlIcebergTable(RefreshTableStmt stmt, Env env) thro
                 stmt.getTableName(), "ICEBERG", icebergProperties, "");
         env.createTable(createTableStmt);
     }
+
+    public void addToRefreshMap(long catalogId, Integer[] sec) {
+        refreshMap.put(catalogId, sec);
+    }
+
+    public void removeFromRefreshMap(long catalogId) {
+        refreshMap.remove(catalogId);
+    }
+
+    public void start() {
+        TaskRefresh taskRefresh = new TaskRefresh();
+        this.refreshScheduler.scheduleAtFixedRate(taskRefresh, 0, REFRESH_TIME,
+                TimeUnit.SECONDS);
+    }
+
+    private class TaskRefresh implements Runnable {
+        @Override
+        public void run() {
+            for (Map.Entry<Long, Integer[]> entry : refreshMap.entrySet()) {
+                Long catalogId = entry.getKey();
+                Integer[] timeGroup = entry.getValue();
+                Integer original = timeGroup[0];
+                Integer current = timeGroup[1];
+                if (current - REFRESH_TIME > 0) {
+                    timeGroup[1] = current - REFRESH_TIME;
+                    refreshMap.put(catalogId, timeGroup);
+                } else {
+                    String catalogName = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogId).getName();
```

Comment:
Catalog may be dropped while you are doing this refresh task.
So need to check if the return value of `getCatalog` is null, or just using `getCatalogOrException` and catch the exception

Reply:
Ok,I will check if the return value of `getCatalog` is null